test/lib/add-redbox-matchers.ts

```diff
@@ -0,0 +1,211 @@
+       * use {@link toDisplayCollapsedRedbox} instead.
+       *
+       * Unintented content in the snapshot should be reported to the Next.js DX team.
+       * <FIXME-internal-frame> in the snapshot would be unintended.
+       * Any node_modules in the snapshot would be unintended.
+       * Differences in the snapshot between Turbopack and Webpack would be unintended.
+       *
+       * @param inlineSnapshot - The snapshot to compare against.
+       */
+      toDisplayRedbox(inlineSnapshot?: string): Promise<void>
+
+      /**
+       * Inline snapshot matcher for a Redbox that's collapsed by default.
+       * When a Redbox is immediately displayed,
+       * use {@link toDisplayRedbox} instead.
+       *
+       * Unintented content in the snapshot should be reported to the Next.js DX team.
+       * <FIXME-internal-frame> in the snapshot would be unintended.
+       * Any node_modules in the snapshot would be unintended.
+       * Differences in the snapshot between Turbopack and Webpack would be unintended.
+       *
+       * @param inlineSnapshot - The snapshot to compare against.
+       */
+      toDisplayCollapsedRedbox(inlineSnapshot?: string): Promise<void>
+    }
+  }
+}
+
+interface RedboxSnapshot {
+  environmentLabel: string
+  label: string
+  description: string
+  source: string
+  stack: string[]
+  count: number
+}
+
+async function createRedboxSnapshot(
+  browser: BrowserInterface
+): Promise<RedboxSnapshot> {
+  const [label, environmentLabel, description, source, stack, count] =
+    await Promise.all([
+      getRedboxLabel(browser),
+      getRedboxEnvironmentLabel(browser),
+      getRedboxDescription(browser),
+      getRedboxSource(browser),
+      getRedboxCallStack(browser),
+      getRedboxTotalErrorCount(browser),
+    ])
+
+  // We don't need to test the codeframe logic everywhere.
+  // Here we focus on the cursor position of the top most frame
+  // From
+  //
+  // pages/index.js (3:11) @ eval
+  //
+  //   1 | export default function Page() {
+  //   2 |   [1, 2, 3].map(() => {
+  // > 3 |     throw new Error("anonymous error!");
+  //     |           ^
+  //   4 |   })
+  //   5 | }
+  //
+  // to
+  //
+  // pages/index.js (3:11) @ Page
+  // > 3 |     throw new Error("anonymous error!");
+  //     |           ^
+  let focusedSource = source
+  if (source !== null) {
+    focusedSource = ''
+    const sourceFrameLines = source.split('\n')
+    for (let i = 0; i < sourceFrameLines.length; i++) {
+      const sourceFrameLine = sourceFrameLines[i]
+      if (sourceFrameLine === '') {
+        continue
+      }
+
+      if (sourceFrameLine.startsWith('>')) {
+        // This is where the cursor will point
+        // Include the cursor and nothing below since it's just surrounding code.
+        focusedSource += '\n' + sourceFrameLine
+        focusedSource += '\n' + sourceFrameLines[i + 1]
+        break
+      }
+      const isCodeFrameLine = /^ {2}\s*\d+ \|/.test(sourceFrameLine)
```

Comment:
Looks cool, for name wise do we plan to use the same util to handle the compiler error (the one with underlined style) as well, or just we want a new util for that?

Reply:
Good point. I think I'd prefer to have them as normal errors at some point so that the codeframe display is normalized. I'd leave this open for now until we get to working on build errors.